core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosUtils.scala

```diff
@@ -0,0 +1,112 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.scheduler.cluster.mesos
+
+import java.util.{List => JList}
+
+import com.google.common.base.Splitter
+import org.apache.mesos.Protos
+import org.apache.mesos.Protos._
+
+import scala.collection.JavaConversions._
+import scala.collection.mutable
+
+
+object MesosUtils {
+
+  /** Helper method to get the key,value-set pair for a Mesos Attribute protobuf */
+  private[mesos] def getAttribute(attr: Attribute) =
+    (attr.getName, attr.getText.getValue.split(',').toSet)
+
+  /** Helper function to pull out a resource from a Mesos Resources protobuf */
+  private[mesos] def getResource(res: JList[Resource], name: String): Double = {
+    for (r <- res if r.getName == name) {
+      return r.getScalar.getValue
+    }
+    0
+  }
+
+  /** Build a Mesos resource protobuf object */
+  private[mesos] def createResource(resourceName: String, quantity: Double): Protos.Resource = {
+    Resource.newBuilder()
+      .setName(resourceName)
+      .setType(Value.Type.SCALAR)
+      .setScalar(Value.Scalar.newBuilder().setValue(quantity).build())
+      .build()
+  }
+
+
+  /**
+   * Match the requirements (if any) to the offer attributes.
+   * if attribute requirements are not specified - return true
+   * else if attribute is defined and no values are given, simple attribute presence is preformed
+   * else if attribute name and value is specified, subset match is performed on slave attributes
+   */
+  private[mesos] def matchesAttributeRequirements(
+    slaveOfferConstraints: Map[String, Set[String]],
+    offerAttributes: Map[String, Set[String]]) =
+    if (slaveOfferConstraints.isEmpty) {
+      true
+    } else {
+      slaveOfferConstraints.forall {
+        // offer has the required attribute and subsumes the required values for that attribute
+        case (name, requiredValues) =>
+          offerAttributes.contains(name) &&
+            // The attributes and their values are case sensitive during comparison
+            // i.e tachyon -> true != Tachyon -> true != tachyon -> True
+            (requiredValues.size == 0 ||
+              requiredValues.forall((v) => offerAttributes(name).contains(v)))
+      }
+    }
+
+  /** Helper method to parse the constraint string */
+  private[mesos] def parseConstraintString(constraintsVal: String): Map[String, Set[String]] = {
+    /*
+      Based on mesos docs:
+      attributes : attribute ( ";" attribute )*
+      attribute : labelString ":" ( labelString | "," )+
+      labelString : [a-zA-Z0-9_/.-]
+    */
+    val splitter = Splitter.on(';').trimResults().withKeyValueSeparator(':')
```

Comment:
I think it would be good to put this in the scaladoc for this method, since it's not clear that the constraints string is the Mesos format of attributes.